cider/function/FunctionLookup.h

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef CIDER_FUNCTION_FUNCTIONLOOKUP_H
+#define CIDER_FUNCTION_FUNCTIONLOOKUP_H
+
+#include <memory>
+#include <optional>
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+#include "function/SubstraitFunctionCiderMappings.h"
+#include "function/substrait/SubstraitFunctionLookup.h"
+#include "function/substrait/SubstraitType.h"
+#include "function/substrait/VeloxToSubstraitMappings.h"
+
+#define SUBSTRAIT_ENGINE "substrait"
+#define PRESTO_ENGINE "presto"
```

Comment:
maybe define these into a ENUM `PLATFORMS?` considering that we may support more platforms.

Reply:
Done, also change FunctionSignature::from_platform from string to enum.